obspy/io/csv/core.py

```diff
@@ -0,0 +1,484 @@
+            except Exception:
+                warn(f'No magnitude found for event {evid}')
+                mag = float('nan')
+                magtype = ''
+                magauthor = ''
+            else:
+                mag = magnitude.mag
+                magtype = magnitude.magnitude_type or ''
+                try:
+                    magauthor = magnitude.creation_info.author
+                except AttributeError:
+                    magauthor = ''
+            try:
+                dep = origin.depth / (1000 if depth_in_km else 1)
+            except Exception:
+                warn(f'No depth set for event {evid}')
+                dep = float('nan')
+            for description in event.event_descriptions:
+                if str(description.type) == 'region name':
+                    region = description.text
+                    break
+            else:
+                region = ''
+            d = {'time': origin.time,
+                 'lat': origin.latitude,
+                 'lon': origin.longitude,
+                 'dep' if depth_in_km else 'depm': dep,
+                 'mag': mag,
+                 'magtype': magtype,
+                 'id': evid,
+                 'author': author,
+                 'contrib': contrib,
+                 'magauthor': magauthor,
+                 'region': region,
+                 }
+            f.write(fmtstr.format(**d).replace('nan', '') + '\n')
+
+
+def load_csv(fname, skipheader=0, only=None, names=None,
+             delimiter=',', **kw):
+    """
+    Load CSV or CSZ file into numpy array
+
+    :param only: sequence, read only columns speified by name
+    :param skipheader, names: see `read_csv`
+    :param **kw: Other kwargs are passed to `np.loadtxt`
+
+    """
+    if isinstance(fname, str) and zipfile.is_zipfile(fname):
+        with zipfile.ZipFile(fname) as zipf:
+            with io.TextIOWrapper(
+                    zipf.open('events.csv'), encoding='utf-8') as f:
+                return load_csv(f)
+    with _open(fname) as f:
+        for _ in range(skipheader):
+            f.readline()
+        if names is None:
+            names = f.readline().strip().split(',')
+        names = _names_sequence(names)
+        dtype = [(n, DTYPE[n]) for n in names if n in DTYPE and
+                 (only is None or n in only)]
+        usecols = [i for i, n in enumerate(names) if n in DTYPE and
+                   (only is None or n in only)]
+        kw.setdefault('usecols', usecols)
+        kw.setdefault('dtype', dtype)
+        return np.genfromtxt(f, delimiter=delimiter, **kw)
+
+
+def events2array(events, **kw):
+    """
+    Convert ObsPy catalog to numpy array
+
+    All kwargs are passed to `load_csv`, e.g. use `only=('lat', 'lon', 'mag')`
+    to get an array with lat, lon, mag parameters.
+    """
+    with io.StringIO() as f:
+        _write_csv(events, f)
+        f.seek(0)
+        return load_csv(f, **kw)
```

Comment:
This is generally useful, but I wonder if it is faster than just parsing the data from the event objects directly? Also, should it live in this module? If we decide to make a generic version of it that doesn't end up writing to csv first its current location doesn't make a lot of sense. 

In light of that, you might consider making it a private function for now.

Reply:
Yes, I agree to turn it to a private function. Actually it should be the other way round, this function implemented in core and as stream method without everything converted to strings and write_csv and cohorts building upon that.
Though, I think the conversion to and from strings here will not increase the running time significantly. Was such a nice low hanging fruit. Related: #1516